docs/content/docs/data-structures/tree.md

```diff
@@ -485,61 +498,98 @@ For the meaning of "simultaneously", see [Types of distributed data structures](
 
 ### Events
 
-The `SharedTree` object supports two events: `beforeChange` and `afterChange`. Your code can create handlers for these events using the utility class `Tree`. See [Tree utility APIs](#tree-utility-apis).
+`SharedTree` supports two node level events: `nodeChanged` and `treeChanged`. Your code can create handlers for these events using the utility class `Tree`. See [Tree utility APIs](#tree-utility-apis).
+
+Additionally, the `TreeView` object includes 3 events that operate over the whole tree. These are `rootChanged`, `afterBatch`, and `commitApplied`.
+
+`rootChanged` fires when the root field (the field that contains the root node) changes. That is, if a new root node is assigned or the schema changes. This will not fire when the node itself changes.
+
+`afterBatch` fires when a batch of changes has been processed by the Fluid Framework and the `TreeView` object is updated.
+
+`commitApplied` fires whenever a local change is applied outside of a transaction or when a local transaction is committed. This is used to get `Revertible` objects to put on the undo stack. See [Undo/Redo support](#undoredo-support) and [Transactions](#transactions).
+
+### Undo/Redo support
```

Comment:
I think we need to explain that undo and redo are unified under the Revertible concept, and we need to give a more detailed explanation of our revert semantics. I imagine we want to keep this section short, so I propose we cover all that in a separate document (a filtered version of the design spec would be a good starting point for that one). This section can then keep things even simpler and boil things down to: "if you want the ability to undo or redo commits, you will need to generate Revertible objects from the "commitApplied" event. See this document for more details about the semantics of undo/redo in SharedTree".